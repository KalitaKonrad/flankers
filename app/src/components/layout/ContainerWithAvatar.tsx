import React, { ComponentProps } from 'react';
import { ImageSourcePropType, StyleSheet, View } from 'react-native';
import { useTheme } from 'react-native-paper';

import { Avatar } from '../shared/Avatar';
import { Container } from './Container';

type ContainerWithAvatarProps = ComponentProps<typeof Container> & {
  avatar: ImageSourcePropType;
  button?: React.ReactNode;
};

const AVATAR_SIZE = 158;

export const ContainerWithAvatar: React.FC<ContainerWithAvatarProps> = (
  props
) => {
  const theme = useTheme();

  return (
    <Container {...props}>
      <View style={styles.avatarContainer}>
        <View
          style={[
            styles.navbarExtension,
            { backgroundColor: theme.colors.primary },
          ]}
        />
        <View style={styles.avatarWrapper}>
          <Avatar
            containerStyle={{ zIndex: 10 }}
            border={4}
            elevation={6}
            size={AVATAR_SIZE}
            src={props.avatar}
          />
          {props.button && <View style={styles.button}>{props.button}</View>}
        </View>
      </View>
      <View style={styles.contentContainer}>{props.children}</View>
    </Container>
  );
};

const styles = StyleSheet.create({
  avatarContainer: {
    alignItems: 'center',
  },
  navbarExtension: {
    width: '100%',
    height: AVATAR_SIZE / 2,
  },
  avatarWrapper: {
    marginTop: -AVATAR_SIZE / 2,
    position: 'relative',
  },
  contentContainer: {
    paddingTop: 24,
  },
  button: {
    elevation: 7,
    position: 'absolute',
    zIndex: 100,
    bottom: 50,
    left: AVATAR_SIZE / 2 + 10,
    backgroundColor: 'red',
  },
});
