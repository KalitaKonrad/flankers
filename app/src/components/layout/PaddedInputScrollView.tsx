import React from 'react';
import { StyleProp, ViewStyle, StyleSheet } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

interface PaddedInputScrollViewProps {
  style?: StyleProp<ViewStyle>;
}

export const PaddedInputScrollView: React.FC<PaddedInputScrollViewProps> = ({
  style,
  children,
}) => {
  return (
    <KeyboardAwareScrollView style={[styles.container, style]}>
      {children}
    </KeyboardAwareScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 8,
    // marginBottom: 110,
    paddingHorizontal: 16,
  },
});
