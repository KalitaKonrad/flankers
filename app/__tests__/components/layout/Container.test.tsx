import { render } from '@testing-library/react-native';
import * as React from 'React';

import { Container } from '../../../src/components/layout/Container';

describe('<Container>', () => {
  it('renders correctly', () => {
    const tree = render(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
