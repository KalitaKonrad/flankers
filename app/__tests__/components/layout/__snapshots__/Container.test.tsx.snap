// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "position": "relative",
        "zIndex": 100,
      },
      undefined,
    ]
  }
/>
`;
